src/common/Compiler.h

```diff
@@ -33,14 +33,63 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #ifndef COMMON_COMPILER_H_
 #define COMMON_COMPILER_H_
 
-// CountTrailingZeroes returns the number of trailing zeroes of the argument in binary.
-// The result is unspecified if the input is 0.
-int CountTrailingZeroes(unsigned int x);
-int CountTrailingZeroes(unsigned long x);
-int CountTrailingZeroes(unsigned long long x);
+// Code making use of compiler intrinsics.
+
+/* CountTrailingZeroes returns the number of
+trailing zeroes of the argument in binary.
+The result is unspecified if the input is 0. */
+#if defined(DAEMON_USE_COMPILER_INTRINSICS) && defined(__GNUC__)
+	inline int CountTrailingZeroes(unsigned int x)
```

Comment:
The formatting seems messed up on all the CountTrailingZeroes definitions

Reply:
Well, I mostly just turned:

```c++
inline int CountTrailingZeroes(unsigned int x) { unsigned long ans; _BitScanForward(&ans, x); return ans; }
```

into:

```c++
inline int CountTrailingZeroes(unsigned int x)
	{ unsigned long ans; _BitScanForward(&ans, x); return ans; }
```

It's just splitting a long line. It's a matter of taste, but otherwise we may just reformat them the non-golfy way.

Reply:
Or is the problem the extra indentation within the `ifdef`?

Reply:
Well if we define a function on multiple lines, the opening brace is normally at the same alignment level as the beginning of the function declaration (`inline int`...). Also being inside a preprocessor block normally isn't a reason to indent non-preprocessor code.

Reply:
Ok